addons/web/static/tests/legacy/fields/basic_fields_tests.js

```diff
@@ -769,6 +769,40 @@ QUnit.module('basic_fields', {
         form.destroy();
     });
 
+    QUnit.test('numeric field: field with type `number` and with keydown on numpad decimal key', async function (assert) {
+        patchWithCleanup(basicFields.NumericField.constructor.prototype, {_onKeydown(ev) {
+            const res = this._super(...arguments);
+            assert.step("Using the numpad decimal did not crash");
+            return res;
+            }})
+
+        this.data.partner.fields.float_field = { string: "Float", type: 'float' };
+        this.data.partner.records[0].float_field = 123;
+
+        const form = await createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: `
+                <form string="Partners">
+                    <field name="float_field" options="{'type': 'number'}"/>
+                </form>
+            `,
+            res_id: 1,
+        });
+
+        // Record edit mode
+        await testUtilsDom.click(form.el.querySelector('.o_form_button_edit'));
+
+        // Get input
+        const floatField = form.el.querySelector('.o_input[name="float_field"]');
+
+        floatField.dispatchEvent(new KeyboardEvent('keydown', { code: 'NumpadDecimal', key: '.' }));
+        // The keyboardevent does not change the value. We cannot verify that the input.value ends vith a '.'
+        // So we have to test the flow and check that it does not crash
+        assert.verifySteps(["Using the numpad decimal did not crash"])
```

Comment:
No need to explain something you don't assert.

It made me realize that the main point of this test is to check that the event is not default prevented!
Remove this and see comment of the patch section.